docs: Edited to remove jargon in Reference, Glossary, and Style sections (#42187)
angular/angular: aio/content/guide/styleguide.md

<div class="s-rule consider">

**Consider** guidelines should generally be followed.

> If you fully understand the meaning behind the guideline and have a good reason to deviate, then do so. Strive to be consistent.

Reviewer: `Strive` is arguably an unnecessary use of an uncommon word here. Perhaps `Aim` instead?

## Application structure and NgModules

> Have a near-term view of implementation and a long-term vision. Start small but keep in mind where the application is heading down the road.

Reviewer: `down the road` is rather idiomatic. If we are striving[sic] to make the docs more accessible to non-native English speakers, should we find a more straightforward phrase here?

> **Why?** Logic may be reused by multiple components when placed within a service and exposed by way of a function.

Reviewer: ```suggestion
**Why?** Logic may be reused by multiple components when placed within a service and exposed as a function.
```
